src/Integration.UnitTests/LocalServices/RuleSetInspectorTests.cs

```diff
@@ -0,0 +1,482 @@
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.CodeDom.Compiler;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace SonarLint.VisualStudio.Integration.UnitTests.LocalServices
+{
+    [TestClass]
+    public class RuleSetInspectorTests
+    {
+        /* Notes: "By default" is referred to the way we create rulesets for projects when binding.
+        Also, please read the comments inside FindConflictingRules to better understand the merge logic (if needed to)
+        */
+        private const int DefaultNumberOfRules = 3;
+
+        private RuleSetInspector testSubject;
+        private ConfigurableServiceProvider serviceProvider;
+        private ConfigurableVsShell shell;
+        private ConfigurableVsGeneralOutputWindowPane outputPane;
+
+        #region Test plumbing
+        public TestContext TestContext { get; set; }
+
+        [TestInitialize]
+        public void TestInitialize()
+        {
+            this.serviceProvider = new ConfigurableServiceProvider();
+
+            this.outputPane = new ConfigurableVsGeneralOutputWindowPane();
+            this.serviceProvider.RegisterService(typeof(SVsGeneralOutputWindowPane), this.outputPane);
+
+            this.shell = new ConfigurableVsShell();
+            this.shell.RegisterPropertyGetter((int)__VSSPROPID2.VSSPROPID_InstallRootDir, () => this.SonarQubeRuleSetFolder);
+            this.serviceProvider.RegisterService(typeof(SVsShell), this.shell);
+           
+            this.testSubject = new RuleSetInspector(this.serviceProvider);
+
+            if (!Directory.Exists(VsRuleSetsDirectory))
+            {
+                Directory.CreateDirectory(VsRuleSetsDirectory);
+            }
+        }
+
+        [TestCleanup]
+        public void TestCleanup()
+        {
+            // Catch release-build issues that would otherwise be ignored because Debug.Assert will not be called
+            this.outputPane.AssertOutputStrings(0); 
+        }
+        #endregion
+
+        #region Properties
+        /// <summary>
+        /// Simulates the solution level SonarQube folder in which we store the fetched rulesets
+        /// </summary>
+        private string SonarQubeRuleSetFolder
+        {
+            get
+            {
+                return this.TestContext.TestRunDirectory;
+            }
+        }
+
+        /// <summary>
+        /// Simulates the usages of the vs ruleset folder for the default rulesets that ship with VS
+        /// </summary>
+        private string VsRuleSetsDirectory
+        {
+            get
+            {
+                return Path.Combine(this.TestContext.TestRunDirectory, RuleSetInspector.DefaultVSRuleSetsFolder);
+            }
+        }
+        #endregion
+
+        #region Tests
+        [TestMethod]
+        public void RuleSetInspector_FindConflictingRules_ProjectLevelOverridesOfTheSolutionRuleset()
+        {
+            using (var temps = new TempFileCollection())
```

Comment:
It doesn't look like there are any rulesets that aren't added to the temp collection. The tests would be cleaner if the clean-up code was moved inside the "CreateXXX" methods and the temp collection disposed during [TestCleanup].